### PR TITLE
Feature/export adjustments cleanup

### LINKED ISSUE #45: Export Adjustments
(Sprint C)
Multiple improvements to the way maps are exported and to the UI controls around the image export. The design is based off the doc ‘MAPS_Exprot_Flow_v6’. Requirements are listed below:
- [x] Move Export to top right corner
- [x] Move Publish to Actions (edit dropdown), and label as "Get Link". Change from full screen display to floating pop-up.
- [x] Move Export options to replace toolbar options when export preview is activated
- [x] Add four controls to top menu bar, destination, size, save, cancel. 
  - Destination 
    - [x] Two options: AVMM and Local
    - [x] Default: AVMM
    - [x] Should be configurable so that dropdown options can be flagged as internal vs. External, etc...
  - Size
    - [x] Sizes: Small, Medium, Large, Custom
    - [x] Bounding box to display solid border when standard size option is selected, size should not be editable
    - [x] Width and Height controls to display when Custom Size is selected (more detail in spec)
  - [x] Save
  - [x] Cancel
- [x] Confirmation pop-up message to appear showing map preview, title, description, and copy-only ID (Bloomberg to populate ID field)
-  ~~Download location description to pop-up when local destination is selected and filetype to default to PNG.~~
-  ~~Map Title to pre-populate as File Name~~

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Closes #45.

Contains all the work from the `feature/export-adjustments-working` working branch, rebased against `bbg_production` and with some commits squashed.

### CODE REVIEW

**Comment by CloudNiner** [comment]
Will take a look as soon as I have a working VM again!

File: lib/assets/javascripts/cdb [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit a2b8834622f2b4d3b849a7e810cff998d982844d
+Subproject commit b308a025d124af7b11144a95adf9bca8e47b7176
```
Comment (CloudNiner) [inline lib/assets/javascripts/cdb]: Looks like this got accidentally updated somehow. If you didn't make any changes to cartodb.js then the removed `a2bb83` commit is the correct one.

**Comment by CloudNiner** [comment]
It looks like the 'Export' button in the top right has the text 'Visualize' until some unknown JS loads on the page. Can we update so that the 'Visualize' text doesn't awkwardly flash before being replaced?

**Comment by CloudNiner** [comment]
+1 after fixing the 'Visualize' text flash. Works great.

**Comment by flibbertigibbet** [comment]
@sharph put in the 'Export' text; any thoughts on that?

**Comment by CloudNiner** [comment]
It's probably a text change in the ruby html template for that header?

**Comment by flibbertigibbet** [comment]
Looks like it's context-dependent; if it's not a visualization, that button is for making a visualization out of it.

**Pull request merged**